network/src/p2p/handler.rs

```diff
@@ -80,14 +82,19 @@ const RETRY_SYNC_MAX: Duration = Duration::from_secs(10); // T1
 const RTT: Duration = Duration::from_secs(10); // T2
 const WAIT_SYNC: Duration = Duration::from_secs(30); // T3 >> T1 + RTT
 
+pub trait ManagingPeerdb: Send + Sync {
+    fn insert(&self, transaction: DBTransaction, key: &[u8], value: &[u8]);
```

Comment:
The arguments are too generic.
Do you have a plan to insert other types except `SocketAddr`?
If not, I think the key type should be `SocketAddr` or `&SocketAddr`.

Reply:
And what does the inserting value mean?
IMO, it is an implementation detail not for an interface.

Reply:
I cannot understand the second one. Can you explain more? 
I mean this one 
https://github.com/CodeChain-io/codechain/pull/1945/commits/916565582681895600fe754c43a0a078f3011e9c#r363035953

Reply:
Currently, the `insert` method takes an rlp encoded socket address as `key` and an rlp encoded time as `value`. The key is necessary; the purpose of this trait is storing the address. I think the purpose of the time is to know how the connection is stable, but I wonder whether it's the role of `ManagingPeerdb`. How to store the address and what address to use are implementation details.

Reply:
Got it and apply you desire changes.